lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java

```diff
@@ -778,6 +780,12 @@ public void deleteFiles(Collection<Path> paths)
                 deletePaths(currentBatch);
             }
         }
+        catch (MultiObjectDeleteException e) {
+            String errors = e.getErrors().stream()
+                    .map(error -> format("key: %s, versionId: %s, code: %s, message: %s", error.getKey(), error.getVersionId(), error.getCode(), error.getMessage()))
+                    .collect(joining(", "));
+            throw new IOException("Exception while batch deleting paths: %s".formatted(errors), e);
```

Comment:
nit: `Exception(s) while deleting paths: %s`

Reply:
I would keep this as-is for consistency with another AmazonClientException catch block.